### PR TITLE
CVSL-203: Generate PDF for PSS  licences

### PR DESCRIPTION
**Pull request by smcveigh941**
- Added licence PDFs for PSS and AP_PSS licence types
- Aligned the numbers in the paragraphs
- Reused paragraphs where possible using `{% include partial %}`, where content differs, just include bespoke paragraphs in each licence type template
- Added print button to top and bottom of print page as per the prototype
- See example of each licence pdf below

[AP_PSS.pdf](https://github.com/ministryofjustice/create-and-vary-a-licence/files/7577218/AP_PSS.pdf)
[AP.pdf](https://github.com/ministryofjustice/create-and-vary-a-licence/files/7577219/AP.pdf)
[PSS.pdf](https://github.com/ministryofjustice/create-and-vary-a-licence/files/7577220/PSS.pdf)

### CODE REVIEW

File: server/views/pages/licence/AP.test.ts [hunk 1]
```diff
@@ -38,7 +38,7 @@ describe('Print an AP licence', () => {
         ],
         bespokeConditions: [{ text: 'Bespoke condition 1' }],
       },
-      additionalConditions: [
```
Comment (smcveigh941) [inline server/views/pages/licence/AP.test.ts:41]: Not a great fan of this test.. not sure its testing anything, other than content. No logic being tested. I left this test in but did not add tests for PSS and AP_PSS licences

File: assets/sass/components/_licence.scss [hunk 2]
```diff
@@ -42,50 +46,51 @@ body {
...
+  .heading {
+    font-weight: bolder;
+  }
 }
 
 .section {
-  clear: left;
-  text-align: justify-all;
+  text-align: justify;
   break-inside: avoid;
```
Comment (timharrison-moj) [inline assets/sass/components/_licence.scss:59]: Thanks for tidying this - its been on my list to come to. What do you think to a pure-CSS footer? Worth pursuing? The Gotenberg footer feels a bit dirty, but it works for now. Something we can come back to I think?

File: server/views/pages/licence/AP.njk [hunk 3]
```diff
@@ -1,39 +1,77 @@
...
-    {{ supervision(licence.licenceStartDate, licence.licenceExpiryDate) }}
-    {{ induction(licence.appointmentPerson, licence.appointmentAddress, licence.appointmentTime, licence.comTelephone) }}
-    {{ releasedToOther() }}
-    {{ conditions(licence.standardLicenceConditions, additionalConditions, licence.bespokeConditions) }}
-    {{ cancellation() }}
-    {{ recall() }}
-    {{ failureToComply() }}
-    {{ signatures(licence.approvedByName, licence.approvedDate, offenderName) }}
-    {{ footer(licence.nomsId, licence.pnc, licence.cro, licence.bookingNo, licence.typeCode, licence.id, licence.version, licence.prisonCode) }}
+    <div id="title" class="title">Licence for {{ offenderName }}</div>
```
Comment (timharrison-moj) [inline server/views/pages/licence/AP.njk:14]: I still have a slight preference for the macro method - creating a simpler top-level document with all sections clearly partitioned into their own files, but as discussed, I'm not precious about it. This change is also clear to follow / maintain and lets see how the various licence types develop, overlap or share sections?

File: server/views/pages/licence/AP.test.ts (hunk 1 above)
Comment (timharrison-moj) [reply server/views/pages/licence/AP.test.ts:41]: It checks that the document structure given in data is rendered into the sections / counts expected, but largely repeating what the integration test does as well.

File: server/views/pages/licence/partials/conditions.njk [hunk 4]
```diff
@@ -1,32 +1,7 @@
...
-{% endmacro %}
-
-{%  macro bespokeConditions(bespokeConditions) %}
-    {% for condition in bespokeConditions %}
-        <div class="condition" id="bespoke-{{ loop.index }}">
-            {{ condition.text }}
+{% macro conditions(conditions) %}
+    {% for condition in conditions %}
+        <div class="condition">
+            <span>{{ condition.text }}</span>
```
Comment (timharrison-moj) [inline server/views/pages/licence/partials/conditions.njk:4]: I like this simplification!

**Review by timharrison-moj — APPROVED** [review]
LGTM.. Very good PR, two more (tidier) licence documents in the bag.

File: assets/sass/components/_licence.scss (hunk 2 above)
Comment (smcveigh941) [reply assets/sass/components/_licence.scss:59]: I havent looked at the footer to be honest! Although I sure it can be done easy enough

**Pull request merged**